src/components/Pagination/Pagination.st.css

```diff
@@ -0,0 +1,205 @@
+
+:import {
+  -st-from: "wix-ui-core/index.st.css";
+  -st-named: Pagination
+}
+
+:import {
+  -st-from: "../../tpaStyles/styleableExt.js";
+  -st-named: wixApply;
+}
+
+:vars {
+  defaultTextFont: "{theme: 'Body-M', size: '16px'}";
+  defaultTextFontColor: color-5;
+  defaultSelectedTextFontColor: color-8;
+  defaultDisabledTextFontColor: color-3;
+}
+
+:vars {
+  TextFont: --overridable;
+  TextFontColor: --overridable;
+  SelectedTextFontColor: --overridable;
+  DisabledTextFontColor: --overridable;
+}
+
+.root {
+  -st-extends: Pagination;
+  -st-states: rtl, paginationMode(enum(pages, input)), navButtonsType(enum(text, arrows)), navButtonsAmount(number);
+  
+  align-items: stretch;
+  box-sizing: border-box;
+  display: inline-flex;
+  font:  wixApply(fallback, wixApply(font, value(TextFont)), wixApply(font, value(defaultTextFont)));
+}
+
+.pageButtonRules {
+  min-width: 1em;
+  min-height: 1em;
+  line-height: 1em;
+  justify-content: center;
+  align-items: center;
+  padding: 0.5em 0.5em;
+  margin: 0.25em;
+}
+
+.root::pageButton {
+  -st-mixin: pageButtonRules;
+  position: relative;
+  color: wixApply(color, wixApply(fallback, value(TextFontColor), value(defaultTextFontColor)));
+}
+
+.root::currentPage {
+  -st-mixin: pageButtonRules;
+  color: wixApply(color, wixApply(fallback, value(SelectedTextFontColor), value(defaultSelectedTextFontColor)));
+  padding: 0.5em;
+}
+
+.root:not(:disabled)::pageButton:hover::before {
+  opacity: 0.2;
+  content: "";
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+}
+
+.root:not(:disabled)::pageButton:hover {
+  opacity: 0.7;
+}
+
+.root:disabled::pageButton {
+  cursor: default;
+}
+
+.root::navButton {
+  min-width: 1em;
+  min-height: 1em;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.root::navButton:disabled, .root[data-preview="disabled"]::navButton {
+  cursor: default;
+}
+
+.root::navButton:disabled svg, .root[data-preview="disabled"]::navButton svg {
+  fill: wixApply(color, wixApply(fallback, value(DisabledTextFontColor), value(defaultDisabledTextFontColor)));
+}
+
+.root::navButton:hover:not(:disabled) {
+  opacity: 0.8;
+}
+
+.root::navButton svg {
+  width: 1em;
+  min-width: 16px;
+  height: 1em;
+  min-height: 16px;
+  fill: wixApply(color, wixApply(fallback, value(TextFontColor), value(defaultTextFontColor)));
+}
+
+.root[dir="rtl"]::navButton svg {
```

Comment:
So regarding the `rtl` prop:
I would prefer having the style **not** be aware of `rtl`.
The `rtl` prop is more for the logic in the `tsx` files, for accessibility for instance.

However, if we have to use `rtl` prop here, then I think it should be used as a stylable state and not using `dir=rtl` (which by the way, I didn't see was added to the element)

Reply:
it's added to the element in [wix-ui-core](https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/components/pagination/Pagination.tsx#L307)